paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSchema.java

```diff
@@ -63,10 +66,15 @@ public MySqlSchema(
                     scale = null;
                 }
                 if (!caseSensitive) {
+                    if (fields.containsKey(fieldName.toLowerCase())) {
+                        LOG.warn(
+                                "Duplicate key '{}' appears in table '{}.{}' when converting fields map keys "
+                                        + "to case-insensitive form. This may cause error.",
+                                fieldName,
+                                databaseName,
+                                originalTableName);
+                    }
                     fieldName = fieldName.toLowerCase();
```

Comment:
```Java
fieldName = fieldName.toLowerCase();
if (fields.containsKey(fieldName)) {
                     ...
}
```

Reply:
I want to retain original `fieldName` in error message.